src/streamlink/plugins/bongacams.py

```diff
@@ -0,0 +1,145 @@
+    "localData" : {
+        "NC_ConnUrl" : validate.url(scheme="rtmp"),
+        "NC_AccessKey" : validate.length(32),
+        "dataKey" : validate.length(32),
+    },
+    "performerData" : {
+        "username" : validate.text,
+    }
+})
+
+
+class AMFMessage2(AMFMessage):
+    def _serialize(self, packet):
+        packet += AMF0String(self.target_uri)
+        packet += AMF0String(self.response_uri)
+        packet += U32BE(AMF0Value.size(self.value))
+        packet += AMF0Value.pack(self.value)
+
+
+class bongacams(Plugin):
+    @classmethod
+    def can_handle_url(self, url):
+        return url_re.match(url)
+
+    def _get_stream_uid(self, username):
+        m = md5(username.encode('utf-8') + str(time.time()).encode('utf-8'))
+        return m.hexdigest()
+
+    def _get_streams(self):
+        match = url_re.match(self.url)
+
+        baseurl = match.group(1)
+        chathost = match.group(2)
+        chathost_url = match.group()
+        amf_gateway_url = urljoin(baseurl, CONST_AMF_GATEWAY_LOCATION)
+        country = urlparse(baseurl).netloc[:2].lower()
+
+        # create http session and set headers
+        http_session = http
+        http_session.headers.update(CONST_HEADERS)
+
+        # get swf url and cookies for next request
+        r = http_session.get(chathost_url, allow_redirects=False)
+
+        if r.is_redirect:
+            self.logger.debug("redirect to: {}", r.headers.get('Location'))
+            raise NoStreamsError
+        if not r.ok:
+            self.logger.debug("Status code for {}: {}", r.url, r.status_code)
+            raise NoStreamsError
+        if len(http_session.cookies) == 0:
+            raise PluginError("Can't get a cookies")
+
+        match = swf_re.search(r.text)
+        if match:
+            swf_url = urljoin(baseurl, match.group())
+            self.logger.debug("swf url found: {}", swf_url)
+        else:
+            swf_url = urljoin(baseurl, CONST_DEFAULT_SWF_LOCATION)
+            self.logger.debug("swf url not found. Will try {}", swf_url)
+
+        # create amf query
+        amf_message = AMFMessage2("svDirectAmf.getRoomData", "/1", [chathost, True])
+        amf_packet = AMFPacket(version=0)
+        amf_packet.messages.append(amf_message)
+
+        # send request and close http-session
+        r = http_session.post(url=amf_gateway_url,
+                              params={CONST_AMF_GATEWAY_PARAM : country},
+                              data=bytes(amf_packet.serialize()))
+        http_session.close()
+
+        if r.status_code != 200:
+            raise PluginError("unexpected status code for {}: {}", r.url, r.status_code)
+
+        amf_response = AMFPacket.deserialize(BytesIO(r.content))
+
+        if len(amf_response.messages) != 1 or amf_response.messages[0].target_uri != "/1/onResult":
+            raise PluginError("unexpected response from amf gate")
+
+        stream_source_info = parse_json(json.dumps(amf_response.messages[0].value), schema=amf_msg_schema)
```

Comment:
FYI, I think you could do `amf_msg_schema.validate(amf_msg_schema)` instead of dumping and parsing again :-)